### PR TITLE
ref(issues/replay): update issue details replay CTA

### LINKED ISSUE #64311: [Design] Iterate on Replay CTA on Issue Details
Synced with @vuluongj20, @rachrwang, @jas-kas, @bruno-garcia and we agreed on the following criteria to update the Replay CTA (both frontend & web backend) on Issue Details:

Requirements:
- Standardize Frontend & Web Backend CTA design.

The frontend design looks like this: 
screenshot_2024-01-31_at_2 18 10___pm_720

And the web backend looks like this:
SCR-20240131-kekv

We want to update the frontend design to look **more like the the new backend design.**

- Instead of show/hide details in the top right, the Replay CTA should be dismissible with an "X" in the top right corner 
  - Dismissing ("X") will stick around for user & project level 
- The current frontend CTA has ability to both "snooze" and "dismiss" -- we'd like to implement both these behaviors for the new design as well.
  - Perhaps show these options with less prominence vs. the "set up now" button
-  After dismissing/snoozing, the whole Session Replay section goes away.

### PR DESCRIPTION
**Pull request by michellewzhang**
Update the Replay CTA (both frontend & web backend) on Issue Details:
- Standardize Frontend & Web Backend CTA design.
- Instead of show/hide details in the top right, the Replay CTA is now dismissible and snoozable with a dropdown menu "X" in the top right corner
   - Dismissing or snoozing will stick around for user & project level
- After dismissing/snoozing, the whole Session Replay section goes away.
- On prod it will show the "View sample replay" button as well.

https://github.com/getsentry/sentry/assets/56095982/a73e6c54-96e3-4ed5-bbe6-d1b4783d4cac

On larger screens:
Screenshot 2024-02-02 at 11 24 50 AM
SCR-20240202-kmgy

On smaller screens we don't display the image because the display gets wonky:
SCR-20240202-kjwc

Closes https://github.com/getsentry/sentry/issues/64311

### CODE REVIEW

File: static/app/components/events/eventReplay/index.spec.tsx [hunk 1]
```diff
@@ -145,30 +145,9 @@ describe('EventReplay', function () {
     });
     render(<EventReplay {...defaultProps} />, {organization});
 
-    expect(await screen.findByText('Configure Session Replay')).toBeInTheDocument();
-  });
-
-  it('should not render the replay inline onboarding component when the project is not JS', function () {
```
Comment (michellewzhang) [inline static/app/components/events/eventReplay/index.spec.tsx:151]: this test isn't applicable now that we're rendering for both frontend & backend

File: static/app/components/events/eventReplay/replayInlineOnboardingPanel.tsx [hunk 2]
```diff
@@ -1,126 +1,178 @@
...
-  const [isHidden, setIsHidden] = useState(() => {
-    const hideUntilTime = getHideUntilTime();
-    if (hideUntilTime && Date.now() < hideUntilTime) {
-      return true;
-    }
-    clearHideUntilTime();
-    return false;
+  const {dismiss, isDismissed} = useDismissAlert({
+    key: LOCAL_STORAGE_KEY,
+    expirationDays: 365,
```
Comment (scttcper) [inline static/app/components/events/eventReplay/replayInlineOnboardingPanel.tsx:44]: this might make more sense as a guide

Comment (michellewzhang) [reply static/app/components/events/eventReplay/replayInlineOnboardingPanel.tsx:44]: @scttcper what do you mean by guide?

Comment (scttcper) [reply static/app/components/events/eventReplay/replayInlineOnboardingPanel.tsx:44]: [these](https://github.com/getsentry/sentry/blob/93560a1b9d01c5bd3e5628ef498daf2205b90198/src/sentry/utils/prompts.py#L4-L19), we can store something in the database so the banner doesn't come back when the browser clears localstorage

Comment (michellewzhang) [reply static/app/components/events/eventReplay/replayInlineOnboardingPanel.tsx:44]: are there any specific benefits to using this vs just local storage? the original replay CTA was using local storage as well

Comment (scttcper) [reply static/app/components/events/eventReplay/replayInlineOnboardingPanel.tsx:44]: it persists for longer and across browsers are the only advantages

Comment (michellewzhang) [reply static/app/components/events/eventReplay/replayInlineOnboardingPanel.tsx:44]: cc @vuluongj20 @Jesse-Box @rachrwang @jas-kas @ryan953 @bruno-garcia

Comment (michellewzhang) [reply static/app/components/events/eventReplay/replayInlineOnboardingPanel.tsx:44]: @scttcper how strongly do you feel about this on a [logaf](https://blog.danlew.net/2020/04/15/the-logaf-scale/) scale?

Comment (ryan953) [reply static/app/components/events/eventReplay/replayInlineOnboardingPanel.tsx:44]: idk how @scttcper feels, but we can always update the implementation later.

Right now we're benefitting from a new design which i'd like to see in the app.

File: static/app/components/events/eventReplay/replayInlineOnboardingPanel.tsx [hunk 3]
```diff
@@ -1,126 +1,178 @@
...
+  border: 1px solid ${p => p.theme.border};
+  border-radius: ${p => p.theme.borderRadius};
+  padding: ${space(2)};
+  margin: ${space(1)} 0;
+  background: linear-gradient(
+    90deg,
+    ${p => p.theme.backgroundSecondary}00 0%,
+    ${p => p.theme.backgroundSecondary}FF 70%,
+    ${p => p.theme.backgroundSecondary}FF 100%
+  );
```
Comment (ryan953) [inline static/app/components/events/eventReplay/replayInlineOnboardingPanel.tsx:137]: subtle 👍

**Review by ryan953 — APPROVED** [review]

File: static/app/components/events/eventReplay/replayInlineOnboardingPanel.tsx (hunk 2 above)
Comment (michellewzhang) [reply static/app/components/events/eventReplay/replayInlineOnboardingPanel.tsx:44]: synced w/ @rachrwang & @bruno-garcia and they don't feel too strongly about it either; might not be worth the extra effort

**Pull request merged**